### PR TITLE
Feature/nearby io fix

### PR DESCRIPTION
**Pull request by ashishkumar468**
## Nearby crashes when Home [From the Nav Drawer] is pressed while nearby is still loading data.

Fixes #{#1846, Nearby crashes when Home [From the Nav Drawer] is pressed while nearby is still loading data } 

## Description (required)
While nearby is still loading, and home from the nav drawer is clicked, the app used to crash with InterruptedIOException.

{Describe the changes made and why they were made.}
For fixing the same, I have catched the exception  the function getFromWikidataQuery(curLatLng, lang, radius) in NearbyPlaces. The cause of the exception could be anything, primarily IO. After catching the exception I returned null for places, for which I had to handle null for places in loadAttractionsFromLocation(LatLng curLatLng).

Tested on {27 & samsung s7}, with { ProdDebug}.
Tested for stability, Nearby no longer crashes in the mentioned scenario.

## Screenshots showing what changed (optional)
 NA

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java [hunk 1]
```diff
@@ -45,7 +45,12 @@ public NearbyPlaces() {
 
             // increase the radius gradually to find a satisfactory number of nearby places
             while (radius <= MAX_RADIUS) {
-                places = getFromWikidataQuery(curLatLng, lang, radius);
+                try {
+                    places = getFromWikidataQuery(curLatLng, lang, radius);
+                }catch (Exception e){
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java]: Please use consistent whitespaces :)

**Comment by misaochan** [comment]
Looks good to me, happy to merge once it has been tested.

**Comment by ashishkumar468** [comment]
@misaochan I have personally tested this 4-5 times. Is there any specific test case you want me to perform ?

**Comment by misaochan** [comment]
@ashishkumar0207 I was referring to additional tests required as part of our project maintenance policy - https://github.com/commons-app/apps-android-commons/wiki/Project-maintenance . All PRs except for versioning/changelogs (including my own!) have to be tested by someone else before they are merged. :)

**Comment by ashishkumar468** [comment]
@misaochan Got you, ofcourse

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java [hunk 2]
```diff
@@ -45,7 +45,12 @@ public NearbyPlaces() {
 
             // increase the radius gradually to find a satisfactory number of nearby places
             while (radius <= MAX_RADIUS) {
-                places = getFromWikidataQuery(curLatLng, lang, radius);
+                try {
+                    places = getFromWikidataQuery(curLatLng, lang, radius);
+                } catch (Exception e) {
+                    Timber.d("exception in fetching nearby places", e.getLocalizedMessage());
+                    return null;
```
[the remark this answers is not in the document]
Comment (ashishkumar468) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java]: should actually have returned places instead, which is anyways Collections.emptyList(), what do you say ?

[the remark this answers is not in the document]
Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java]: Yup, my point was not to return null as the function is not annotated as `@Nullable`.

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java [hunk 3]
```diff
@@ -45,7 +45,12 @@ public NearbyPlaces() {
 
             // increase the radius gradually to find a satisfactory number of nearby places
             while (radius <= MAX_RADIUS) {
-                places = getFromWikidataQuery(curLatLng, lang, radius);
+                try {
+                    places = getFromWikidataQuery(curLatLng, lang, radius);
+                } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (ashishkumar468) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java]: @maskaravivek Yes actually that would be better, did the required changes

**Pull request merged**